packages/lib-react-components/src/AnimatedNumber/AnimatedNumber.js

```diff
@@ -4,78 +4,104 @@ import { interpolate } from '@visx/vendor/d3-interpolate'
+  }
+  return new Promise((resolve) => {
+    select(element)
       .data([value])
       .transition()
       .duration(duration)
       .textTween(() => {
         const interpolator = interpolate(initialValue, value)
         return t => {
           const interpolatedValue = interpolator(t)
-          if (interpolatedValue === value) setAnimated(true) // animation complete!
+          if (interpolatedValue === value) {
+            resolve() // animation complete!
+          }
           const niceValue = formatValue(interpolatedValue)
           return niceValue
         }
       })
-  }
+  })
+}
 
-  function lessAnimation() {
-    select(numRef.current)
-    .data([value])
-    .transition()
-    .duration(0)
-    .textTween(() => {
-      return () => {
-        setAnimated(true) // animation complete!
-        return formatValue(value)
-      }
-    })
+/**
+ * Use d3 to tween `element` from `initialValue` to `value`.
+ * @param {HTMLElement} element 
+ * @param {number} initialValue
+ * @param {number} value 
+ * @returns a Promise that resolves once the animation is complete.
+ */
+function lessAnimation(element, initialValue, value) {
+  if (value === initialValue) {
+    return
   }
+  return new Promise((resolve) => {
+    select(element)
+      .data([value])
+      .transition()
+      .duration(0)
+      .textTween(() => {
+        return () => {
+          resolve() // animation complete!
+          return formatValue(value)
+        }
+      })
+  })
+}
 
-  function formatValue(num) {
-    return format(',d')(num)
-  }
+function AnimatedNumber({ duration = 1000, value }) {
+  const numRef = useRef(null)
+  const initialValueRef = useRef(0)
+  const initialValue = initialValueRef.current
+  const [animated, setAnimated] = useState(false)
 
   useEffect(() => {
     // If we already animated the number once, don't observe intersection
     // This could happen if the value prop updates, but page did not refresh
-    if (animated) return
+    if (animated || value === initialValue) return
 
     const numElement = numRef.current
 
-    const intersectionObserver = new window.IntersectionObserver(entries => {
+    const intersectionObserver = new window.IntersectionObserver(async entries => {
       // If intersectionRatio is 0, the target is out of view and we do not need to do anything.
       if (entries[0].intersectionRatio <= 0) return
 
       // Once target element is in viewport, animate it then unobserve
-      if (!prefersReducedMotion && !animated) {
-        animateValue()
-        intersectionObserver.unobserve(numElement)
-      } else {
-        lessAnimation()
-        intersectionObserver.unobserve(numElement)
-      }
+      const animationFunction =
+        !prefersReducedMotion && !animated ? animateValue : lessAnimation
+      await animationFunction(numElement, initialValue, value, duration)
+      setAnimated(true)
+      initialValueRef.current = value
+      intersectionObserver.unobserve(numElement)
```

Comment:
Statements in JavaScript have side effects, but expressions don't, so I changed the logic to use a ternary expression rather than a if statement. See [Rethinking the JavaScript Ternary Operator](https://jrsinclair.com/articles/2021/rethinking-the-javascript-ternary-operator/) by James Sinclair.

Reply:
The side effect, in the original code, was animating the content of a DOM node outside of the React render cycle. I guess that's still happening here, but the `await` might be easier to read.